### PR TITLE
New Components section

### PR DESCRIPTION
**Pull request by robdodson**
Hey fellas,

Here's my first stab at the new Components section. This contains the following changes.

- Creates a new top-level section called Building Components.
- Moves the Custom Elements and Shadow DOM primers out of the Getting Started section, and into Building Components.
- Moves the HowTo: Components out of the Architecture section and into the Building Components section. Their section is now called Examples.
- Creates a new Best Practices doc (similar to the PWA checklist).
- Adds redirects for the old docs to their new home.

Sorry for the rush on this but I'd love to get this PR landed (maybe even launched?) before Polymer Summit which is on the 22nd of August.

### CODE REVIEW

**Comment by robdodson** [comment]
blargh, fixing test failures. hold please.......

File: src/content/en/fundamentals/building-components/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: Components are the building blocks of modern web applications. What best practices should you follow when building your own components so they can stand the test of time?
+
+{# wf_updated_on: 2017-08-14 #}
+{# wf_published_on: 2017-08-14 #}
+
```
Comment (ebidel) [inline src/content/en/fundamentals/building-components/index.md]: add in `{# wf_blink_components: Blink>DOM #}`

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+  Aim to keep primitive data attributes and properties in sync. Reflecting from
+  property to attribute, and vice versa.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  You never know how a user will try to interact with your element. They might
+  set a property in JavaScript, and then expect to be able to read that value
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: Delete 'to be able'.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+  using an API like <code>getAttribute</code>. If every attribute has a
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: Make sure method names have () at the end of them.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 4]
```diff
@@ -0,0 +1,535 @@
...
+  corresponding property, and both of them reflect, it will make it easier for
+  users to work with your element. In other words, calling
+  <code>setAttribute('foo', value)</code> should also set a corresponding
+  <code>foo</code> property. And vice versa. There are, of course, exceptions to
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: Change 'property. And vice versa' to 'property and vice versa'.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 5]
```diff
@@ -0,0 +1,535 @@
...
+  this rule. You shouldn't reflect high frequency properties, e.g.
+  <code>currentTime</code> in a video player. Use your best judgment. If it
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: method?

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 6]
```diff
@@ -0,0 +1,535 @@
...
+  seems like a likely a user will interact with a property or attribute, and
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: Delete 'likely'.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 7]
```diff
@@ -0,0 +1,535 @@
...
+  it's not burdensome to reflect it, then do so.
+      </td>
+    </tr>
+    <tr>
+      <td><b>Example</b></td>
+      <td>
+  The <a href="/web/fundamentals/building-components/examples/howto-checkbox">
+  <code>&lt;howto-checkbox&gt;</code></a> element. Further explained in
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: 'Further explained' to 'This is further explained'.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 8]
```diff
@@ -0,0 +1,535 @@
...
+      <th colspan=2>
+        <h3>
+  Aim to only accept rich data (objects, arrays) as properties.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Generally speaking, there are no example of built-in HTML Elements that accept
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: 'example' to 'examples'

elements should be lower case

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 9]
```diff
@@ -0,0 +1,535 @@
...
+  Do not reflect rich data properties to attributes.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Reflecting rich data properties to attributes is needlessly expensive,
+  requiring serializing and deserializing the same JavaScript objects. Unless
+  you have a specific use case that this feature solves, it's probably best to
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: Perhaps change to 'Unless you have a use case that can only be solved with this feature,'?

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 10]
```diff
@@ -0,0 +1,535 @@
...
+        <h3>
+  Consider checking for properties that may have been set before the element
+  upgraded.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  A developer may attempt to set a property on the element before its
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: 'A developer may' to 'You may'

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 11]
```diff
@@ -0,0 +1,535 @@
...
+  Dispatch events in response to internal component activity.
+        </h3>
+      </th>
+    </tr>
+    <tr>
+      <td><b>Why?</b></td>
+      <td>
+  Your component may have properties that change in response to activity that
+  only your component knows about. For example, if a timer or animation
+  finishes, or a resource has finished loading. It's helpful to dispatch events
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: 'has finished' to 'finishes'

Also, this needs to be a complete sentence.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 12]
```diff
@@ -0,0 +1,535 @@
...
+    this.setAttribute('role', 'checkbox');
+  if (!this.hasAttribute('tabindex'))
+    this.setAttribute('tabindex', 0);
+```
+
+### Make properties lazy {: #lazy-properties }
+
+A developer might attempt to set a property on your element before its
+definition has been loaded. This is especially true if the developer is using a
+framework which handles loading components, stamping them to the page, and
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: What is 'stamping them to the page'? Is there a better term to use here?

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 13]
```diff
@@ -0,0 +1,535 @@
...
+binding their properties to a model.
+
+```html
+<!--
+Here Angular is declaratively binding its model's isChecked property to the
+checkbox's checked property. If the definition for howto-checkbox was lazy
+loaded it's possible that Angular might attempt to set the checked property
+before the element has upgraded.
+-->
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: This doesn't really need to be a comment. If you want to make the connection clearer, start with 'In the following example' or something like that.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 14]
```diff
@@ -0,0 +1,535 @@
...
+### Make properties lazy {: #lazy-properties }
+
+A developer might attempt to set a property on your element before its
+definition has been loaded. This is especially true if the developer is using a
+framework which handles loading components, stamping them to the page, and
+binding their properties to a model.
+
+```html
+<!--
+Here Angular is declaratively binding its model's isChecked property to the
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: `isChecked` not isChecked (formatting)

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 15]
```diff
@@ -0,0 +1,535 @@
...
+checkbox's checked property. If the definition for howto-checkbox was lazy
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: `checked` attribute

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 16]
```diff
@@ -0,0 +1,535 @@
...
+loaded it's possible that Angular might attempt to set the checked property
+before the element has upgraded.
+-->
+<howto-checkbox [checked]="defaults.isChecked"></howto-checkbox>
+```
+
+A Custom Element should handle this scenario by checking if any properties have
+already been set on its instance. The `<howto-checkbox>` demonstrates this
+pattern using a method called `_upgradeProperty`.
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: Parenthesis on the method

Check throughout

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 17]
```diff
@@ -0,0 +1,535 @@
...
+  if (this.hasOwnProperty(prop)) {
+    let value = this[prop];
+    delete this[prop];
+    this[prop] = value;
+  }
+}
+```
+
+`_upgradeProperty` captures the value from the unupgraded instance and deletes
+the property so it does not shadow the Custom Element's own property setter.
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/best-practices.md]: 'custom element's' should be lower case.

File: src/content/en/fundamentals/building-components/examples/howto-tooltip.md [hunk 18]
```diff
@@ -84,8 +65,7 @@ limitations under the License.
 </div>
 
 
-<script src="https://cdn.rawgit.com/webcomponents/custom-elements/master/custom-elements.min.js"></script>
-<script src="https://cdn.rawgit.com/webcomponents/shadydom/master/shadydom.min.js"></script>
+<script src="https://cdn.rawgit.com/webcomponents/webcomponentsjs/d5b7ca65/webcomponents-sd-ce.js"></script>
```
Comment (petele) [inline src/content/en/fundamentals/building-components/examples/howto-tooltip.md]: You can make the script error go away by indenting this script block by at least one space.

File: src/content/en/fundamentals/building-components/customelements.md [hunk 19]
```diff
@@ -13,29 +13,43 @@ description: Custom elements allow web developers to define new HTML tags, exten
 ### TL;DR {: #tldr .hide-from-toc }
 
 With [Custom Elements][spec], web developers can **create new HTML tags**,
-beef-up existing HTML tags, or extend the components other developers have authored.
-The API is the foundation of [web components](http://webcomponents.org/){: .external }. It brings
-a web standards-based way to create reusable components using nothing more than
-vanilla JS/HTML/CSS. The result is less code, modular code, and more reuse in our apps.
+beef-up existing HTML tags, or extend the components other developers have
+authored. The API is the foundation of [web
+components](http://webcomponents.org/){: .external }. It brings a web
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/customelements.md]: We no longer use the {: .external } macro. Please fix throughout.

File: src/content/en/fundamentals/building-components/customelements.md [hunk 20]
```diff
@@ -13,29 +13,43 @@ description: Custom elements allow web developers to define new HTML tags, exten
...
-
-The browser gives us an excellent tool for structuring web applications.
-It's called HTML.  You may have heard of it! It's declarative, portable, well supported, and easy to work with. Great as HTML may be, its vocabulary and extensibility are limited. The [HTML living standard](https://html.spec.whatwg.org/multipage/){: .external } lacks a way to automatically associate JS behavior with your markup... until now.
-
-Custom elements are the answer to modernizing HTML; filling in the missing pieces,
-and bundling structure with behavior. If HTML doesn't provide the solution to a problem,
-we can create a custom element that does. **Custom elements teach the browser new tricks while preserving the benefits of HTML**.
+Note: This article describes the new <a
+href="https://html.spec.whatwg.org/multipage/scripting.html#custom-elements"
+target="_blank">Custom Elements v1 spec</a>. If you've been using custom
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/customelements.md]: Make sure the spec name matches the actual spec exactly.

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md [hunk 21]
```diff
@@ -0,0 +1,616 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+
+{# wf_updated_on: 2017-08-14 #}
+{# wf_published_on: 2017-04-06 #}
+
```
Comment (petele) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: should this include the `wf_blink_components` tag?

File: src/content/en/fundamentals/building-components/customelements.md [hunk 22]
```diff
@@ -13,29 +13,43 @@ description: Custom elements allow web developers to define new HTML tags, exten
...
+elements, chances are you're familiar with the <a
+href="https://www.chromestatus.com/features/4642138092470272">v0 version shipped
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/customelements.md]: Change 'v0 version' to 'version 0'

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md [hunk 23]
```diff
@@ -0,0 +1,616 @@
...
+    font-size: 20px;
+    margin-left: 8px;
+  }
+</style>
+
+<howto-checkbox aria-labelledby="join-label"></howto-checkbox>
+<span id="join-label">Join Newsletter</span>
+
+
+<script src="https://cdn.rawgit.com/webcomponents/webcomponentsjs/d5b7ca65/webcomponents-sd-ce.js"></script>
```
Comment (petele) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: Script errors go away if you indent at least one space.

File: src/content/en/fundamentals/building-components/customelements.md [hunk 24]
```diff
@@ -13,29 +13,43 @@ description: Custom elements allow web developers to define new HTML tags, exten
...
+in Chrome 33</a>. The concepts are the same, but the v1 spec has important API
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/customelements.md]: 'version 1', not 'v1'

File: src/content/en/fundamentals/building-components/examples/howto-tooltip.md [hunk 25]
```diff
@@ -1,7 +1,7 @@
 project_path: /web/_project.yaml
 book_path: /web/fundamentals/_book.yaml
 
-{# wf_updated_on: 2017-07-25 #}
+{# wf_updated_on: 2017-08-14 #}
 {# wf_published_on: 2017-04-06 #}
```
Comment (petele) [inline src/content/en/fundamentals/building-components/examples/howto-tooltip.md]: should this include the `wf_blink_components` tag?

File: src/content/en/fundamentals/building-components/examples/index.md [hunk 26]
```diff
@@ -2,7 +2,7 @@ project_path: /web/_project.yaml
 book_path: /web/fundamentals/_book.yaml
 description: "HowTo: Components"
 
-{# wf_updated_on: 2017-06-12 #}
+{# wf_updated_on: 2017-08-14 #}
 {# wf_published_on: 2017-04-06 #}
```
Comment (petele) [inline src/content/en/fundamentals/building-components/examples/index.md]: should this include the `wf_blink_components` tag?

File: src/content/en/fundamentals/building-components/customelements.md [hunk 27]
```diff
@@ -13,29 +13,43 @@ description: Custom elements allow web developers to define new HTML tags, exten
...
+differences. Keep reading to see what's new or check out the section on <a
+href="#historysupport">History and browser support</a> for more info.
+
+The browser gives us an excellent tool for structuring web applications. It's
+called HTML.  You may have heard of it! It's declarative, portable, well
+supported, and easy to work with. Great as HTML may be, its vocabulary and
+extensibility are limited. The [HTML living
+standard](https://html.spec.whatwg.org/multipage/){: .external } lacks a way to
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/customelements.md]: 'lacks a way' to 'has always lacked a way'

File: src/content/en/fundamentals/building-components/customelements.md [hunk 28]
```diff
@@ -13,29 +13,43 @@ description: Custom elements allow web developers to define new HTML tags, exten
...
+automatically associate JS behavior with your markup... until now.
+
+Custom elements are the answer to modernizing HTML; filling in the missing
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/customelements.md]: Change the semi-colon to a comma.

File: src/content/en/fundamentals/building-components/customelements.md [hunk 29]
```diff
@@ -269,48 +325,41 @@ or create an instance in JavaScript:
...
+  been moved into a new <code>document</code> (e.g. someone called
+  <code>document.adoptNode(el)</code>).</td> </tr> </tbody> </table>
+
+The browser calls the `attributeChangedCallback()` for any attributes
+whitelisted in the `observedAttributes` array (see [Observing changes to
+attributes](#attrchanges)). Essentially, this is a performance optimization.
+When users change a common attribute like `style` or `class`, you don't want to
+be spammed with tons of callbacks.
 
 **Reaction callbacks are synchronous**. If someone calls `el.setAttribute(...)`
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/customelements.md]: I know this isn't part of the PR, but can you please remove the '...'.

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md [hunk 30]
```diff
@@ -0,0 +1,616 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+
+{# wf_updated_on: 2017-08-14 #}
+{# wf_published_on: 2017-04-06 #}
+
+# HowTo: Components – howto-checkbox {: .page-title }
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: This title needs to include the '<' and '>', but do NOT add backticks.

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md [hunk 31]
```diff
@@ -0,0 +1,616 @@
...
+{% include "web/_shared/contributors/robdodson.html" %}
+{% include "web/_shared/contributors/surma.html" %}
+
+<link rel="stylesheet" href="main.css">
+
+## Summary {: #summary }
+
+A `<howto-checkbox` represents a boolean option in a form. The most common type
+of checkbox is a dual-type which allows the user to toggle between two
+choices -- checked and unchecked.
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: Use a colon or an emdash.

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md [hunk 32]
```diff
@@ -0,0 +1,616 @@
...
+[What can ARIA do?][what-aria] and [Using tabindex][using-tabindex].
+
+When the checkbox is checked, it adds a `[checked]` boolean attribute, and sets
+a corresponding `checked` property to `true`. In addition, the element sets an
+`[aria-checked]` attribute to either `"true"` or `"false"`, depending on its
+state. Clicking on the checkbox with a mouse, or space bar, toggles these
+checked states.
+
+The checkbox also supports a `disabled` state. If either the `disabled` property
+is set to true or the `[disabled]` attribute is applied, the checkbox sets
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: Why is 'disabled' in square brackets? Please remove them.

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md [hunk 33]
```diff
@@ -0,0 +1,616 @@
...
+`aria-disabled="true"` and set `tabindex="-1"`.
+
+## Reference {: #reference }
+
+- [Checkbox pattern in ARIA Authoring Practices 1.1][checkbox-pattern]
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: 'pattern' should be upper case.

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md [hunk 34]
```diff
@@ -0,0 +1,616 @@
...
+- [What can ARIA do?][what-aria]
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: Check how the original capitalizes this title. The last word should always be upper case regardless of length or part of speech.

File: src/content/en/fundamentals/building-components/index.md [hunk 35]
```diff
@@ -0,0 +1,96 @@
...
+<div class="attempt-left">
+  <h2>Custom Elements</h2>
+  <a href="/web/fundamentals/building-components/customelements">
+    <div class="align-center">
+      <img width="88" src="/web/images/md-icons/ic_code_black_24px.svg">
+    </div>
+  </a>
+  <p>
+    Custom Elements give developers the ability to extend HTML and create their
+    own tags. Because Custom Elements are standards based they benefit from the
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/index.md]: 'Custom elements' should be lower case.

File: src/content/en/fundamentals/building-components/index.md [hunk 36]
```diff
@@ -0,0 +1,96 @@
...
+<div class="attempt-right">
+  <h2>Shadow DOM</h2>
+  <a href="/web/fundamentals/building-components/shadowdom">
+    <div class="align-center">
+      <img width="88" src="/web/images/md-icons/ic_border_style_black_24px.svg">
+    </div>
+  </a>
+  <p>
+    Shadow DOM is a web standard that offers component style and markup
+    encapsulation. It is critically important piece in the Web Components story
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/index.md]: 'It is critically' to 'It is a critically'

'piece in' to 'piece of'

File: src/content/en/fundamentals/building-components/index.md [hunk 37]
```diff
@@ -0,0 +1,96 @@
...
+    as it ensures that a component will work in any environment, regardless of
```
Comment (jpmedley) [inline src/content/en/fundamentals/building-components/index.md]: Remove the comma.

File: src/content/en/fundamentals/building-components/best-practices.md [hunk 38]
```diff
@@ -0,0 +1,535 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: A checklist which breaks down all the things we think it takes to be a well behaved Custom Element.
```
Comment (robdodson) [inline src/content/en/fundamentals/building-components/best-practices.md]: quick question. What about Shadow DOM? Should that be 'shadow DOM' throughout as well?

File: src/content/en/fundamentals/building-components/examples/howto-tooltip.md (hunk 18 above)
Comment (robdodson) [inline src/content/en/fundamentals/building-components/examples/howto-tooltip.md]: @surma wanted to put this on your radar. Maybe we could add it to our build script?

File: src/content/en/fundamentals/building-components/best-practices.md (hunk 5 above)
Comment (robdodson) [inline src/content/en/fundamentals/building-components/best-practices.md]: Not sure I understand this comment?

File: src/content/en/fundamentals/building-components/best-practices.md (hunk 10 above)
Comment (robdodson) [inline src/content/en/fundamentals/building-components/best-practices.md]: Changed this to:

"A developer using your component may"

I'm trying to refer to when you build a component and then give it to somebody else on your team who uses it in a way you maybe didn't expect.

File: src/content/en/fundamentals/building-components/best-practices.md (hunk 14 above)
Comment (robdodson) [inline src/content/en/fundamentals/building-components/best-practices.md]: since this is inside of an HTML comment I'm not sure if the backticks will work but I can try

File: src/content/en/fundamentals/architecture/building-components/index.md [hunk 39]
```diff
@@ -0,0 +1,97 @@
...
+{# wf_blink_components: Blink>DOM #}
+
+# Building Components {: .page-title }
+
+Components are the building blocks of modern web applications. What best
+practices should you follow when building your own components so they can stand
+the test of time?
+
+<div class="attempt-left">
+  <h2>Custom Elements</h2>
```
Comment (petele) [inline src/content/en/fundamentals/architecture/building-components/index.md]: nit: wrap Custom Elements in a hyperlink

File: src/content/en/fundamentals/architecture/building-components/index.md [hunk 40]
```diff
@@ -0,0 +1,97 @@
...
+    Custom Elements give developers the ability to extend HTML and create their
+    own tags. Because Custom Elements are standards based they benefit from the
+    Web's built-in component model. The result is more modular code that can be
+    reused in many different contexts.
+    <a href="/web/fundamentals/architecture/building-components/customelements">Learn more</a>
+  </p>
+</div>
+
+<div class="attempt-right">
+  <h2>Shadow DOM</h2>
```
Comment (petele) [inline src/content/en/fundamentals/architecture/building-components/index.md]: nit: wrap Shadow DOM in a hyperlink.

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md (hunk 34 above)
Comment (robdodson) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: The original is "What can ARIA do", should I change that as well? I think since it's a h2 subheading, it uses lowercase

File: src/content/en/fundamentals/building-components/examples/howto-checkbox.md (hunk 32 above)
Comment (robdodson) [inline src/content/en/fundamentals/building-components/examples/howto-checkbox.md]: Removing them.

The reason we originally used them is because square brackets denote a CSS attribute selector. So if folks are familiar with that convention and see `[foo]` it helps to reinforce that we mean a foo="" attribute, instead of a js property. I think we started doing this in our element comments and it just kinda stuck. But i'm happy to undo it if it's not WF convention.

**Comment by robdodson** [comment]
I think one of the only remaining issues is regarding the web components logo on the index page. It looks a little big on mobile and is off center on tablet. Is there a recommended way to handle these images?

image

File: src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md [hunk 41]
```diff
@@ -0,0 +1,617 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+
+{# wf_updated_on: 2017-08-14 #}
+{# wf_published_on: 2017-04-06 #}
+{# wf_blink_components: Blink>DOM #}
+
+# HowTo: Components – <howto-checkbox> {: .page-title }
```
Comment (petele) [inline src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md]: You cannot have "markdown" in a heading. You can use `HowTo: Components - &lt;howto-checkbox&gt;`. Encoding the entities will STILL FAIL the test, but I've verified it works and I'll fix the tests asap.

File: src/content/en/fundamentals/architecture/building-components/examples/howto-tooltip.md [hunk 42]
```diff
@@ -1,10 +1,11 @@
 project_path: /web/_project.yaml
 book_path: /web/fundamentals/_book.yaml
 
-{# wf_updated_on: 2017-07-25 #}
+{# wf_updated_on: 2017-08-14 #}
 {# wf_published_on: 2017-04-06 #}
+{# wf_blink_components: Blink>DOM #}
 
-# HowTo: Components – howto-tooltip {: .page-title }
+# HowTo: Components – <howto-tooltip> {: .page-title }
```
Comment (petele) [inline src/content/en/fundamentals/architecture/building-components/examples/howto-tooltip.md]: You cannot have "markdown" in a heading. You can use `HowTo: Components - &lt;howto-checkbox&gt;`. Encoding the entities will STILL FAIL the test, but I've verified it works and I'll fix the tests asap.

File: src/content/en/fundamentals/index.md [hunk 43]
```diff
@@ -10,22 +10,48 @@ nav.devsite-page-nav, .devsite-rating-container, .page-title {display:none;}
...
+  <a href="engage-and-retain/push-notifications/" class="button button-primary">
+    Get Started
+  </a>
+</div>
 
-The Push API and Notification API open a whole new set of
-possibilities for you to re-engage with your users.
+<div class="attempt-right">
+  <h2>Building Web Components</h2>
+  <img src="/web/images/common/webcomponents-logo.png" class="attempt-right">
```
Comment (petele) [inline src/content/en/fundamentals/index.md]: Two options for handling this.
1. Put your own custom responsive class on it
2. Ignore for now, this page will get a refresh in the next 4-6 weeks.

File: src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md (hunk 41 above)
Comment (petele) [inline src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md]: PS - this is fixed in #4946

File: src/content/en/fundamentals/architecture/building-components/examples/howto-tooltip.md (hunk 42 above)
Comment (petele) [inline src/content/en/fundamentals/architecture/building-components/examples/howto-tooltip.md]: PS - this is fixed in #4946

File: src/content/en/fundamentals/index.md (hunk 43 above)
Comment (robdodson) [inline src/content/en/fundamentals/index.md]: Let's go with 2 :)

**Comment by robdodson** [comment]
I _think_ everything is now included. @petele PTAL

File: src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md [hunk 44]
```diff
@@ -0,0 +1,636 @@
...
+          // focusability from an element.
+          // Elements with `tabindex=-1` can still be focused with
+          // a mouse or by calling `focus()`.
+          // To make sure an element is disabled and not focusable, remove the
+          // `tabindex` attribute.
+          if (hasValue) {
+            this.removeAttribute('tabindex');
+            // If the focus is currently on this element, unfocus it by
+            // calling the `HTMLElement.blur()` method.
+            if (document.activeElement === this)
```
Comment (valdrinkoshi) [inline src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md]: This check won't work if `howto-checkbox` is contained in a shadowRoot (`document.activeElement` will be the host of that shadowRoot)
Is it bad to just invoke `this.blur()`?

Comment (robdodson) [inline src/content/en/fundamentals/architecture/building-components/examples/howto-checkbox.md]: Would you mind moving this to https://github.com/GoogleChrome/howto-components
